app/controllers/api/zones_controller.rb

```diff
@@ -1,4 +1,32 @@
 module Api
   class ZonesController < BaseController
+    INVALID_ZONES_ATTRS = ID_ATTRS + %w[created_on updated_on].freeze
+
+    # Edit an existing zone. Certain fields are meant for internal use only
+    # and may not be edited. Attempting to edit one of the forbidden fields
+    # will result in a bad request error.
+    #
+    def edit_resource(type, id, data)
+      bad_attrs = data_includes_invalid_attrs(data)
+
+      if bad_attrs.present?
+        msg = "Attribute(s) '#{bad_attrs}' should not be specified for updating a zone resource"
+        raise BadRequestError, msg
+      end
+
+      super
+    end
+
+    private
```

Comment:
I am a bit concerned about this being the same as this:

https://github.com/ManageIQ/manageiq-api/pull/690/files#diff-a8c5eb840e76e0a1632c7acdb2b03a43R20

Well also the part above.

I think that the behavior should be moved to a/the? parent class or a mixin.

We can agree on merging this (and the 3rd one) and then refactor it once the functionality is there. If you prefer it that way.

WDYT?

Reply:
The fields actually aren't identical - created_at vs created_on. But yes, we should eventually try to put this into some common code.